src/lib/auth.ts

```diff
@@ -58,58 +64,75 @@ export const ROLES: {
 };
 
 export const DEVICE_API_KEY_PERMISSIONS = [
-	`resin.device_type.get?describes_device/any(d:d/${matchesActor})`,
-	`resin.device.get?${matchesActor}`,
+	'resin.device_type.read?describes_device/canAccess()',
+	`resin.device.read?${matchesActor}`,
 	`resin.device.update?${matchesActor}`,
-	`resin.supervisor_release.get?should_manage__device/any(d:d/${matchesActor})`,
-	`resin.application.read?${ownsDevice} or depends_on__application/any(da:da/${ownsDevice})`,
-	`resin.application_tag.read?application/any(a:a/${ownsDevice} or a/depends_on__application/any(da:da/${ownsDevice}))`,
-	`resin.device_config_variable.get?device/any(d:d/${matchesActor} or d/belongs_to__application/any(a:a/depends_on__application/any(da:da/${ownsDevice})))`,
+	'resin.supervisor_release.read?should_manage__device/canAccess()',
+	'resin.application.read?owns__device/canAccess() or depends_on__application/canAccess()',
+	'resin.application_tag.read?application/canAccess()',
+	'resin.device_config_variable.read?device/canAccess()',
 	`resin.device_config_variable.set?device/any(d:d/${matchesActor})`,
-	`resin.device_tag.get?device/any(d:d/${matchesActor})`,
+	'resin.device_tag.read?device/canAccess()',
 	`resin.device_tag.set?device/any(d:d/${matchesActor})`,
-	`resin.application_config_variable.get?${applicationControlsDevice}`,
-	`resin.release.read?should_be_running_on__device/any(d:d/${matchesActor}) or ${belongsToApplicationOrIsManagedByDevice}`,
-	`resin.release_tag.read?release/any(r:r/should_be_running_on__device/any(d:d/${matchesActor}) or r/${belongsToApplicationOrIsManagedByDevice})`,
-	`resin.device_environment_variable.get?device/any(d:d/belongs_to__application/any(a:a/depends_on__application/any(da:da/${ownsDevice})))`,
-	`resin.device_environment_variable.all?device/any(d:d/${matchesActor})`,
-	`resin.application_environment_variable.get?${applicationControlsDevice}`,
+	'resin.application_config_variable.read?application/canAccess()',
+	'resin.release.read?should_be_running_on__device/canAccess() or belongs_to__application/canAccess()',
+	'resin.release_tag.read?release/canAccess()',
+	'resin.device_environment_variable.read?device/canAccess()',
+	...writePerms(
+		'resin.device_environment_variable',
+		`device/any(d:d/${matchesActor})`,
+	),
+	'resin.application_environment_variable.read?application/canAccess()',
 
 	// Dependent device permissions
+	'resin.device.read?belongs_to__application/any(a:a/depends_on__application/canAccess())',
```

Comment:
Updated this as well, @afitzek just to double check it's fine to have a nested `canAccess` like this right?

Reply:
I think this should work as intended: You can read the device resource, if there exists a `belongs_to__application` that has a `depends_on__application` that is readable by the device.